plugins/catalog-react/api-report.md

```diff
@@ -524,12 +524,24 @@ export type UnregisterEntityDialogProps = {
 };
 
 // @public
-export function useEntity<T extends Entity = Entity>(): {
-  entity: T;
+export function useAsyncEntity<
+  T extends Entity = Entity,
+>(): UseAsyncEntityResponse<T>;
+
+// @public
+export interface UseAsyncEntityResponse<T> {
```

Comment:
Realized this type already exists, I think we should use `EntityLoadingStatus` instead. We can give it an entity type parameter that defaults to `Entity` for compatibility.

Gotta say I somewhat dislike the lack of runtime guarantee that the entity is actually of type `T`. Another route we could take, especially with the async hook, is to just skip the type parameter for now.